File: profiles/go/template/internal/hello/hello.go
// Package hello is a sample package shipped with the starter.
// Replace its contents (or remove it entirely) once real code lives here.
package hello

import (
	"context"
	"errors"
	"fmt"
	"strings"
)

// ErrEmptyName is returned when Greet is called with an empty or
// whitespace-only name.
var ErrEmptyName = errors.New("hello: name must not be empty")

// Greet returns a greeting for the given name. Returns ErrEmptyName if name
// is empty or whitespace-only. Respects context cancellation.
func Greet(ctx context.Context, name string) (string, error) {
	if err := ctx.Err(); err != nil {
		return "", fmt.Errorf("greet: %w", err)
	}
	trimmed := strings.TrimSpace(name)
	if trimmed == "" {
		return "", ErrEmptyName
	}
	return fmt.Sprintf("Hello, %s!", trimmed), nil
}

File: profiles/java/template/src/main/java/{{ java_package | replace('.', '/') }}/Hello.java
package {{ java_package }};

import java.util.Objects;

/** Sample value-producing class shipped with the starter. Replace once real code lives here. */
public final class Hello {

    private Hello() {
        // utility class
    }

    /**
     * Returns a greeting for the given name.
     *
     * @param name target of the greeting; must be non-null and non-blank
     * @return formatted greeting
     * @throws NullPointerException if {@code name} is null
     * @throws IllegalArgumentException if {@code name} is blank
     */
    public static String greet(String name) {
        Objects.requireNonNull(name, "name");
        String trimmed = name.strip();
        if (trimmed.isEmpty()) {
            throw new IllegalArgumentException("name must not be blank");
        }
        return "Hello, " + trimmed + "!";
    }
}
